x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/FunctionRegistry.java

```diff
@@ -220,12 +221,12 @@ public FunctionDefinition resolveFunction(String name) {
     }
 
     public String concreteFunctionName(String alias) {
-        String normalized = normalize(alias);
-        return aliases.getOrDefault(normalized, normalized);
+        String upperCase = alias.toUpperCase(Locale.ROOT);
+        return aliases.getOrDefault(upperCase, upperCase);
     }
 
     public boolean functionExists(String name) {
-        return defs.containsKey(normalize(name));
+        return defs.containsKey(name);
```

Comment:
This should use the `concreteFunctionName` since one might ask for a lower_case function name or alias.